Added Target Acquisition for Turrets

Zexenek opened the pull request:
Turrets lock on targets until it is dead or out of range before changing targets.

Also optimised the search script to run in intervals instead of every frame to improve performance.

Zexenek committed:
Added Target Acquisition for Turrets

Turrets lock on targets until it is dead or out of range before changing targets.

Also optimised the search script to run in intervals instead of every frame to improve performance.

relinkz commented on CuddleTreeDefenseGame/Assets/Scripts/Tools.cs:
```diff
@@ -41,16 +41,16 @@ public static GameObject[] GetChildren(GameObject parent)
         return children;
     }
-    public static Collider2D FindNearestTarget(Vector2 originPosition, string tag, float minTurretRange, float maxTurretRange)
+    public static Collider2D FindNearestTarget(GameObject originPosition, string tag, float minTurretRange, float maxTurretRange)
```
gameObj, namnet är förvirrande

relinkz commented on CuddleTreeDefenseGame/Assets/Scripts/Tools.cs:
```diff
@@ -59,4 +59,13 @@ public static Collider2D FindNearestTarget(Vector2 originPosition, string tag, f
         }
         return nearestTarget;
     }
+    public static float? IsInRange(GameObject originObject, GameObject targetObject, float minRange, float maxRange)
```
GameObject to position

relinkz commented on CuddleTreeDefenseGame/Assets/Scripts/Turret.cs:
```diff
@@ -1,60 +1,95 @@
+            if(nearestTarget != null)
+            {
+                target = nearestTarget.gameObject;
+                yield return null;
```
remove

Zexenek committed:
Added Range indicator when mouse hover over Tower

Simple circle based on the tower range is displayed when hovering over a tower.

Also refactored previous code for readability as per suggestions.
